src/EA.Weee.Web/Controllers/OrganisationRegistrationController.cs

```diff
@@ -249,45 +248,71 @@ public async Task<ActionResult> RegisterSmallProducer(string searchTerm)
         [HttpGet]
         public ActionResult Type()
         {
-            return View(new ExternalOrganisationTypeViewModel());
+            return View(new OrganisationTypeViewModel());
         }
 
         [HttpPost]
         [ValidateAntiForgeryToken]
-        public async Task<ActionResult> Type(ExternalOrganisationTypeViewModel model)
+        public ActionResult Type(OrganisationTypeViewModel model)
         {
             if (ModelState.IsValid)
             {
                 var organisationType = model.SelectedValue.GetValueFromDisplayName<ExternalOrganisationType>();
-                var routeValues = new { organisationType = model.SelectedValue, };
-
-                await transactionService.CaptureData(User.GetAccessToken(), model);
+                var routeValues = new { organisationType = model.SelectedValue };
 
                 switch (organisationType)
                 {
                     case ExternalOrganisationType.SoleTrader:
-                        return RedirectToAction("Index", "Holding", routeValues);
+                        return RedirectToAction(nameof(SoleTraderDetails), "OrganisationRegistration", routeValues);
                     case ExternalOrganisationType.Partnership:
-                        return RedirectToAction("Index", "Holding", routeValues);
+                        return RedirectToAction(nameof(PartnershipDetails), "OrganisationRegistration", routeValues);
                     case ExternalOrganisationType.RegisteredCompany:
-                        return RedirectToAction("Index", "Holding", routeValues);
+                        return RedirectToAction(nameof(RegisteredCompanyDetails), "OrganisationRegistration", routeValues);
                 }
             }
 
             return View(model);
         }
 
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<ActionResult> RegisteredCompanyDetails(RegisteredCompanyDetailsViewModel model)
+        {
+            if (!ModelState.IsValid)
+            {
+                var countries = await GetCountries();
+
+                model.Address.Countries = countries;
+
+                return View(model);
+            }
+
+            await transactionService.CaptureData(User.GetAccessToken(), model);
```

Comment:
@stevelee-civica We capture the model detail using the organisation transaction service.